### ISSUE TITLE
install_folder usage in CMakeDefinitionsBuilder can break custom generators

### ISSUE THREAD

**Issue by mjvk** [conan-io/conan#5568]
To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
Conan 1.18
Windows 10 x86_64
Custom generator such as https://github.com/pepe82sh/ConanCmakeToolchainGenerator
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

#5021 moves install_folder outside the check for the generators. When we use a custom generator such as https://github.com/pepe82sh/ConanCmakeToolchainGenerator the CMakeDefinitionsBuilder cannot be used to get the cmake definitions.

This results in:
conanfile.txt: ERROR: Generator CmakeToolchain(file:conan_toolchain.cmake) failed
'ConanFile' object has no attribute 'install_folder'
ERROR: 'ConanFile' object has no attribute 'install_folder'

I'm not sure if this is the custom generator misusing CMakeDefininitionsBuilder, or the CMakeDefininitionsBuilder using install_folder when it should not.

I think the functionality this custom generator brings are great and I am willing to have a look to fix this in a good way (on conan or CmakeToolchain), but for now I am clueless :-).

**Comment by memsharded** [comment]
Able to reproduce it with:

```
git clone  https://github.com/pepe82sh/ConanCmakeToolchainGenerator
cd ConanCmakeToolchainGenerator
conan create . user/testing
```

**Comment by memsharded** [comment]
Hi @mjvk 

This shouldn't be considered a bug, because you are using internal private components of Conan which are not documented nor stable :)

However, as in the context of a generator it seems reasonable that the generator itself should have the ``install_folder`` defined. I am submitting a PR that could fix this. How urgent it is? Why might try to release a 1.18.1 patch if necessary.

What I really like is your generator. We are currently reviewing Conan approach to integrate build systems and we definitely want to generate a toolchain, just like you are doing. We will certainly take some ideas and learning from it, thanks very much for sharing!!

**Comment by mjvk** [comment]
All credits go to @pepe82sh ! No hurry it is possible to circumvent it by just setting an empty install folder.

**Comment by pepe82sh** [comment]
The generator was an idea that emerged in #2956. Thanks for the mension @mjvk !

**Comment by memsharded** [comment]
This has been merged, will be released in 1.19
